apps/login/src/app/login/route.ts

```diff
@@ -66,6 +66,19 @@ async function loadSessions({
   return response?.sessions ?? [];
 }
 
+export function constructUrl(request: NextRequest, path: string) {
+  const forwardedHost =
+    request.headers.get("x-zitadel-forward-host") ??
+    request.headers.get("host");
+  const basePath = process.env.NEXT_PUBLIC_BASE_PATH || "";
+  return new URL(
+    `${basePath}${path}`,
+    forwardedHost?.startsWith("http")
```

Comment:
the host never starts with "http", right?

Why don't we use the x-forwarded-proto and fallback to request.protocol?

the protocol field looks like this:
![image](https://github.com/user-attachments/assets/1cd79bb5-ee36-4c04-9e74-49a10885c633)

the x header looks like this:
![image](https://github.com/user-attachments/assets/5d62f8b9-31f8-4057-9a65-22593dfed471)

Reply:
isn't this highly inconsistent? request.protocol states `https:` and x-forwarded-proto `https`.

Reply:
have a look at this. do you think this is better? 
https://github.com/zitadel/typescript/pull/383/commits/56cd9ce453c231aa2b4d8d2173cae1440c920822